### PR TITLE
Update to work with discord.js v13

### PR DESCRIPTION
**Pull request by ExtremeDevelopments**
This PR adds support for discord.js v13 along with adding some typings updates for TypeScript users and updates all node modules to their latest release.

### CODE REVIEW

**Comment by ExtremeDevelopments** [comment]
Not done

File: package.json [hunk 1]
```diff
@@ -19,24 +19,25 @@
   "license": "MIT",
   "private": false,
   "dependencies": {
-    "discord.js": "^12.5.1",
-    "node-fetch": "^2.6.1",
+    "cluster": "^0.7.7",
```
Comment (DevYukine) [inline package.json:22]: cluster is a build in module, the one on npm is 10 years old and is not something we would wanna use

File: src/Sharding/ShardClientUtil.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { Client, Util } from 'discord.js';
-import { ClusterIPC } from '../IPC/ClusterIPC';
-import { IPCEvents } from '../Util/Constants';
+import { Client, Util } from 'discord.js'
```
Comment (DevYukine) [inline src/Sharding/ShardClientUtil.ts:1]: please do not remove semicolons from imports

File: src/Cluster/BaseCluster.ts [hunk 3]
```diff
@@ -25,11 +25,21 @@ export abstract class BaseCluster {
 	public async init() {
 		const shardUtil = this.client.shard! as ShardClientUtil;
 		await shardUtil.init();
-		this.client.once('ready', () => shardUtil.send({ op: IPCEvents.READY, d: this.id }, { receptive: false }));
-		this.client.on('shardReady', id => shardUtil.send({ op: IPCEvents.SHARDREADY, d: { id: this.id, shardID: id } }, { receptive: false }));
-		this.client.on('shardReconnecting', id => shardUtil.send({ op: IPCEvents.SHARDRECONNECT, d: { id: this.id, shardID: id } }, { receptive: false }));
-		this.client.on('shardResume', (id, replayed) => shardUtil.send({ op: IPCEvents.SHARDRESUME, d: { id: this.id, shardID: id, replayed } }, { receptive: false }));
-		this.client.on('shardDisconnect', ({ code, reason, wasClean }, id) => shardUtil.send({ op: IPCEvents.SHARDDISCONNECT, d: { id: this.id, shardID: id, closeEvent: { code, reason, wasClean } } }, { receptive: false }));
+		this.client.once('ready', () => {
```
Comment (DevYukine) [inline src/Cluster/BaseCluster.ts:28]: this whole change is not needed and style only

File: src/Cluster/Cluster.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
-import { Worker, fork } from 'cluster';
+import cluster, { Worker } from 'cluster';
+const fork = cluster.fork
```
Comment (DevYukine) [inline src/Cluster/Cluster.ts:2]: please import fork from cluster directly in the above import statement, no extra const needed

File: src/IPC/MasterIPC.ts [hunk 5]
```diff
@@ -2,10 +2,10 @@ import { EventEmitter } from 'events';
 import { Server, NodeMessage } from 'veza';
 import { Util } from 'discord.js';
 import { ShardingManager } from '..';
-import { isMaster } from 'cluster';
+import cluster from 'cluster';
 import { IPCEvents, SharderEvents } from '../Util/Constants';
 import { IPCRequest } from './ClusterIPC';
-
+const isPrimary = cluster.isPrimary
```
Comment (DevYukine) [inline src/IPC/MasterIPC.ts:8]: please import isPrimary from cluster directly in the above import statement, no extra const needed

File: src/Sharding/ShardingManager.ts [hunk 6]
```diff
@@ -1,10 +1,12 @@
-import { Client, ClientOptions } from 'discord.js';
+import { Client, ClientOptions, Intents } from 'discord.js';
 import { MasterIPC } from '../IPC/MasterIPC';
 import { Cluster } from '../Cluster/Cluster';
 import { http, SharderEvents } from '../Util/Constants';
 import { EventEmitter } from 'events';
 import { cpus } from 'os';
-import { isMaster, setupMaster } from 'cluster';
+import cluster from 'cluster';
+const isPrimary = cluster.isPrimary
```
Comment (DevYukine) [inline src/Sharding/ShardingManager.ts:8]: please import isPrimary from cluster directly in the above import statement, no extra const needed

File: src/Sharding/ShardingManager.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
...
+const setupPrimary = cluster.setupPrimary
```
Comment (DevYukine) [inline src/Sharding/ShardingManager.ts:9]: please import setupPrimary from cluster directly in the above import statement, no extra const needed

**Review by DevYukine — CHANGES_REQUESTED** [review]
otherwise LGTM

**Review by ExtremeDevelopments — COMMENTED** [review]
Done

**Review by ExtremeDevelopments — COMMENTED** [review]
Done 2

**Review by ExtremeDevelopments — COMMENTED** [review]
Done

**Review by DevYukine — APPROVED** [review]
LGTM

**Pull request merged**